.github/workflows/ci.yaml

```diff
@@ -43,18 +43,32 @@ jobs:
       - id: build-snap
         name: Build snap
         uses: snapcore/action-build@v1
-        with:
-          snapcraft-channel: 7.x/candidate
 
       - name: Upload built snap job artifact
         uses: actions/upload-artifact@v3
         with:
           name: opensearch_snap_amd64
           path: "opensearch_*.snap"
 
+      - name: Set snap release channel
+        run: |
+          version="$(cat snap/snapcraft.yaml | yq .version)"
+          channel="${version%%.*}/edge"
+          
+          echo "version=${version}" >> $GITHUB_ENV
+          echo "channel=${channel}" >> $GITHUB_ENV
+
+      - name: Publish snap to Store
+        uses: snapcore/action-publish@v1
+        env:
+          SNAPCRAFT_STORE_CREDENTIALS: ${{ secrets.STORE_LOGIN }}
+        with:
+          snap: opensearch_${{env.version}}_amd64.snap
+          release: ${{env.channel}}/core2404
```

Comment:
**question**: Is this just a placeholder for the channel? Once the snap is fully tested with the charm, the `*edge/core24` would just become `*/edge` no?

Also also, probably `core24` is the correct format instead of `core2404` (no "04" on snap core)

Reply:
I believe both `core22` and `core24` need to be published. And yes it should be renamed to `core24` in my opinion.

Reply:
Only `core24` would be needed. The snap will download whatever base it needs, so I'm not sure releasing two different bases is useful.